docs/getting-started/installing/installing-core-linux.md

```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
 
 ## Run your App
 
-You need to restore packages for your app, based on your project.json, with `dnu restore`.
+You need to restore packages for your app, based on your project.json, with `dotnet restore`.
 
 ```console
-dnu restore
-dnx run
+dotnet restore
+dotnet run
 
 Hello, Linux
-Love from CoreCLR.
+Love from .NET Core.
+
+```
+
+## Compile your application
 
+Running from source is great for rapid prototyping and trying out things. However, in due time you will want to actually compile your application to get increase in speed and similar benefits. In order to do that, we will use `dotnet compile` command that will produce a runnable executable for our Hello World app.
+
+While you're still in the application's directory, type
+
+    dotnet compile
+    
+This will produce a `bin` directory in your directory. The structure of the drop path is `./bin/[configuration]/[framework]/`. Configuration refers to either *Release* or *Debug*, while framework is essentially a framework ID (i.e. dnxcore50). Inside this directory there will be several files, the most important of which is the binary that will have the same name as your application. Running this will give us the message we saw in the previous example. 
+
+**Note**: this binary requires a shared runtime to be installed on the machine. If you wish to create a self-contained application that includes the runtime and that you can just copy over to another machine, you will need to use `dotnet publish` command or compile your application 
+
+
+## Create a single native binary 
+
+Finally, let's exercise a new feature that we've added to our .NET Command Line Interface: producing single native binaries. These binaries do not require a shared runtime to work; you can just copy the single file over to another Ubuntu machine and just run it. 
+
+The process is pretty similar to the above, with the addition of one more switch. 
+
+```console
+dotnet restore
+dotnet compile --native
 ```
 
-## Building .NET Core from source
+After the compile command finishes, we can just run the resulting binary. By convention, the compile command drops the results in ./bin/[configuration]/[framework]/native/[binary name]. Running this binary will get us our greeting! 
+
+> **Note**
+> This capability is still in its infancy. Therefore, only the simplest of programs will be able to be natively compiled. 
+
+## Building .NET Core runtime from source
+
+.NET Core is an open source project that is hosted on GitHub. This means that you can, at any given time, clone the repository and build .NET Core from source. This is a more advanced scenario that is usually used when you want to add features to the .NET runtime or the BCL or if you are a contributor to these projects. The detailed instruction on how to build .NET Core windows can be found in the [Build .NET Core on Linux](https://github.com/dotnet/coreclr/blob/master/Documentation/building/linux-instructions.md) on GitHub.
```

Comment:
.NET Core is comprised of multiple repos - CoreCLR, CoreFX, Roslyn, CoreRT, etc. The above phrasing should be updated to help explain the right concepts to our development community. It will be great if we can link to a PPT/Article that explains what .NET Core is.